src/Core/Grid/Definition/Factory/CartRuleGridDefinitionFactory.php

```diff
@@ -0,0 +1,262 @@
+                ],
+            ])
+            ->setAssociatedColumn('name')
+            )
+            ->add((new Filter('priority', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Priority', [], 'Admin.Global'),
+                ],
+            ])
+            ->setAssociatedColumn('priority')
+            )
+            ->add((new Filter('code', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Code', [], 'Admin.Global'),
+                ],
+            ])
+            ->setAssociatedColumn('code')
+            )
+            ->add((new Filter('quantity', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Quantity', [], 'Admin.Catalog.Feature'),
+                ],
+            ])
+            ->setAssociatedColumn('quantity')
+            )
+            ->add((new Filter('date_to', DateRangeType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Expiration date', [], 'Admin.Catalog.Feature'),
+                ],
+                'date_format' => 'YYYY-MM-DD HH:mm:ss',
+            ])
+            ->setAssociatedColumn('date_to')
+            )
+            ->add((new Filter('active', YesAndNoChoiceType::class))
+            ->setAssociatedColumn('active')
+            )
+            ->add((new Filter('actions', SearchAndResetType::class))
+            ->setAssociatedColumn('actions')
+            ->setTypeOptions([
+                'reset_route' => 'admin_common_reset_search_by_filter_id',
+                'reset_route_params' => [
+                    'filterId' => self::GRID_ID,
+                ],
+                'redirect_route' => 'admin_cart_rules_index',
+            ])
+            ->setAssociatedColumn('actions')
+            )
+            ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getBulkActions(): BulkActionCollection
+    {
+        return (new BulkActionCollection())
+            ->add((new SubmitBulkAction('enable_selection'))
+            ->setName($this->trans('Enable selection', [], 'Admin.Actions'))
+            ->setOptions([
+                'submit_route' => 'admin_cart_rules_bulk_enable_status',
+            ])
+              )
+            ->add((new SubmitBulkAction('disable_selection'))
+            ->setName($this->trans('Disable selection', [], 'Admin.Actions'))
+            ->setOptions([
+                'submit_route' => 'admin_cart_rules_bulk_disable_status',
+            ])
+              )
+            ->add((new SubmitBulkAction('delete_selection'))
```

Comment:
Or here?
🤔

Reply:
yes, same here.
@JevgenijVisockij you can use `BulkDeleteActionTrait` for that. Once again you can check another grids in develop like Manufacturer for reference.